wlroots/wlr_types/box.py

```diff
@@ -3,6 +3,20 @@
 from wlroots import ffi
 
 
+def _int_getter(attr):
+    def getter(self):
+        return getattr(self, "_" + attr)
+    return getter
+
+
+def _int_setter(attr):
+    def setter(self, value):
+        if not isinstance(value, int):
+            raise ValueError(attr + " should be an int")
+        setattr(self, "_" + attr, value)
+    return setter
```

Comment:
For the `Box`, should these be getting and setting the `self._ptr.*` values, rather than the `self._*` values? Currently, I think setting for example `box.x` will just change `box._x`, not `box._ptr.x`.

If it makes sense to change to assigning to `_ptr`, then when we are assigning into an int in a cffi struct, I don't think the explicit type check is necessary, cffi will complain if you try to assign an invalid type into the value.

Reply:
Huh, well this explains why my window borders weren't rendering correctly!